src/anemoi/training/losses/mse.py

```diff
@@ -23,69 +24,87 @@ class WeightedMSELoss(nn.Module):
 
-        self.register_buffer("weights", node_weights, persistent=True)
-        if data_variances is not None:
-            self.register_buffer("ivar", data_variances, persistent=True)
+        self.register_buffer("node_weights", node_weights, persistent=True)
+        if feature_weights is not None:
+            self.register_buffer("feature_weights", feature_weights, persistent=True)
 
     def forward(
         self,
         pred: torch.Tensor,
         target: torch.Tensor,
         squash: bool = True,
+        feature_indices: torch.Tensor | None = None,
+        feature_scale: bool = True,
     ) -> torch.Tensor:
         """Calculates the lat-weighted MSE loss.
 
         Parameters
         ----------
         pred : torch.Tensor
-            Prediction tensor, shape (bs, lat*lon, n_outputs)
+            Prediction tensor, shape (bs, (optional_ensemble), lat*lon, n_outputs)
         target : torch.Tensor
-            Target tensor, shape (bs, lat*lon, n_outputs)
+            Target tensor, shape (bs, (optional_ensemble), lat*lon, n_outputs)
         squash : bool, optional
             Average last dimension, by default True
+        feature_indices:
+            feature indices (relative to full model output) of the features passed in pred and target
+        feature_scale:
+            If True, scale the loss by the feature_weights
 
         Returns
         -------
         torch.Tensor
             Weighted MSE loss
-
         """
+        if pred.ndim == 4:
+            pred = pred.mean(dim=1)
+
+        torch.save(self.node_weights, "node_weights.pt")
+        torch.save(pred, "pred.pt")
+        torch.save(target, "target.pt")
+
         out = torch.square(pred - target)
 
-        # Use variances if available
-        if hasattr(self, "ivar"):
-            out *= self.ivar
+        if feature_scale and hasattr(self, "feature_weights"):
+            out = (
+                out * self.feature_weights
+                if feature_indices is None
+                else out * self.feature_weights[..., feature_indices]
+            )
 
         # Squash by last dimension
         if squash:
             out = self.avg_function(out, dim=-1)
             # Weight by area
-            out *= self.weights.expand_as(out)
-            out /= self.sum_function(self.weights.expand_as(out))
+            out *= self.node_weights.expand_as(out)
+            out /= self.sum_function(self.node_weights.expand_as(out))
             return self.sum_function(out)
 
-        # Weight by area
-        out *= self.weights[..., None].expand_as(out)
+        # Weight by area, due to weighting construction is analagous to a mean
+        out *= self.node_weights[..., None].expand_as(out)
         # keep last dimension (variables) when summing weights
-        out /= self.sum_function(self.weights[..., None].expand_as(out), axis=(0, 1, 2))
+        out /= self.sum_function(self.node_weights[..., None].expand_as(out))
```

Comment:
Why do we remove the axis here?

Reply:
axis is referencing all dimensions of the tensor, there is no need